### ISSUE TITLE
Expo-video-thumbnails fails to get thumbnails from videos copied from iOS messages

### ISSUE THREAD

**Issue by ChristopherGabba** [expo/expo#25601]
### Minimal reproducible example

https://snack.expo.dev/@christophergabba/expo-clipboard-thumbnail-snack

### Summary

The generate thumbnail function works great for everything except videos copied from SMS messages on iOS. 

Steps to reproduce: 
1.) Copy video from any random SMS text chain with a friend (hold finger and then press "Copy" from sub menu)
2.) Open expo snack above and press the "Generate Thumbnail" button. 

Logs will state: 
```
Error: "Calling the 'getThumbnail' function has failed
→ Caused by: File 'file:///var/mobile/Library/SMS/Attachments/87/07/19753090-DD5D-4610-BEB2-F603CA76FF8B/video_1_ed4ed7140fb845af958b1845b9dbfc35.MP4' is not readable" in Error: Calling the 'getThumbnail' function has failed << → Caused by: File 'file:///var/mobile/Library/SMS/Attachments/87/07/19753090-DD5D-4610-BEB2-F603CA76FF8B/video_1_ed4ed7140fb845af958b1845b9dbfc35.MP4' is not readable <<     at i (expo-video.thumbnails:3:7293) <<     at t (expo-video.thumbnails:3:14816)
```

I know the file is readable because the video plays perfectly on expo-av's video. If I take that file path and drop it into an expo-video, it plays perfectly:

```
 <Video 
    source={{uri: "file:///var/mobile/Library/SMS/Attachments/87/07/19753090-DD5D-4610-BEB2-F603CA76FF8B/video_1_ed4ed7140fb845af958b1845b9dbfc35.MP4"}}
    shouldPlay={true}
    style={$container}
    onError={onError}
 />
```

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.0
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.7.0 - /opt/homebrew/bin/node
      npm: 10.1.0 - /opt/homebrew/bin/npm
      Watchman: 2023.09.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.9862592
      Xcode: 15.0.1/15A507 - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: ^0.10.7 => 0.10.7 
      expo: ^49.0.7 => 49.0.16 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.7 => 0.72.7 
      react-native-web: ~0.19.6 => 0.19.9 
    npmGlobalPackages:
      eas-cli: 5.6.0
      expo-cli: 6.3.10
    Expo Workflow: managed

**Comment by ChristopherGabba** [comment]
Please leave this issue open. Thanks

**Comment by fobos531** [comment]
@ChristopherGabba Is there anything special about that specific video that you're having issues with? Where did it originate from? Did it come from another app? I tried this on my end and I was unable to reproduce the issue.

**Comment by ChristopherGabba** [comment]
@fobos531 There is nothing special from the video. 

Exact steps to reproduce on iOS:

1. Open up expo snack above.
2. Move expo snack into background and go to any video in iMessage
3. Hold finger on video in iMessage and hit "copy"
4. Move back to expo-snack and hit the "generate thumbnail" button (which uses expo-clipboard to paste the local url)
5. See error alert

Here is a reproduction video using a video from iMessage:

https://github.com/expo/expo/assets/122328438/9cf5c1bb-3e88-4d21-b441-009ea5a8eb11

**Comment by ChristopherGabba** [comment]
@fobos531 Good morning, haven't heard from anyone in a while, just checking in... 
Thanks again, Chris

**Comment by ChristopherGabba** [comment]
Just keeping this open and not stale.

**Comment by ChristopherGabba** [comment]
Should not be stale, still an acitive issue

**Comment by joedurand-hub** [comment]
I have the same problem

**Comment by ChristopherGabba** [comment]
Posted over a year ago with a perfectly reproducible example and nothing lol....

**Comment by ChristopherGabba** [comment]
Bumping this again. This is still an issue.

**Comment by ChristopherGabba** [comment]
Not stale, still an active issue with an easy repro.

**Comment by ChristopherGabba** [comment]
@hirbod another >1 year old, super reproducible issue that is right up your alley! 🥹🥹

**Comment by jlo1** [comment]
Started seeing this issue after updating to expo-53, while using the `react-native-create-thumbnail` library.
Switched to `expo-video-thumbnails` to see if it'd help, but also saw an error creating thumbnails with this library too.

Note: only seeing the issue with my .mp4 files, but not with my .mov files

```
Error attaching file: Error: Calling the 'getThumbnail' function has failed
→ Caused by: File 'file:///var/mobile/Containers/Data/Application/..../tmp/ScreenRecording_07-09-2025%252011-01-24_1.mp4' is not readable
    at construct (native)
```

**Comment by hirbod** [comment]
I can try to look into this, but in the meantime, have you tried the new [generateThumbnailsAsync](https://docs.expo.dev/versions/latest/sdk/video/#generatethumbnailsasynctimes-options) directly implemented in `expo-video`? Unlike `expo-video-thumbnails`, it can also accept an array of times and thus extracts data much faster if you need many thumbnails at once. You can also pass more options to resize images on the fly.

I'm curious if the same issue happens there. Please let me know.

**Comment by ChristopherGabba** [comment]
@hirbod I have not yet, but will now. Only downside I can see to this is that it requires making a video player to call the thumbnails function. May not be the most in-line function friendly approach. Still doable.

Maybe you call like so?
```tsx
const player = createVideoPlayer(url)
player.generateThumbnailsAsync()
player.release()
```

**Comment by ChristopherGabba** [comment]
Okay @hirbod,

I updated the expo-snack above to try the `expo-video` package and it really doesn't serve as a good replacement that I can tell.

I tried it with a working url:
```tsx
const BAD_BUNNY_TEST_VIDEO_URL =
  'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4';

        const player = createVideoPlayer({
          uri: BAD_BUNNY_TEST_VIDEO_URL,
          contentType: 'auto',
          assetId: 'ABCD',
          useCaching: false,
        });

       const thumbnail = await player.generateThumbnailsAsync(1000);

```

This method actually does not work for any url. I was able to get it to work using the `const player = useVideoPlayer(BAD_BUNNY_TEST_VIDEO_URL)` hook and then calling `player.generateThumbnailsAsync` worked. 

So there really isn't a great way to run this in line within a function without an active `useVideoPlayer` method. 

Note that this method still failed when using a local copied uri from a text chain:

ex: "file:///var/mobile/Library/SMS/Attachments/56/06/BE070D47-704F-41DD-A4A2-86297E3F1363/ScreenRecording_07-11-2025%2013-15-24_1.mov"

TLDR:
Neither method works for getting a thumbnail from an iOS message.

**Comment by ChristopherGabba** [comment]
Not stale

**Comment by gasolin** [comment]
updated snack example to Expo 54 https://snack.expo.dev/@gasolin/expo-clipboard-thumbnail-54

@ChristopherGabba can you still reproduce that?

**Comment by gasolin** [comment]
I have a sample video (any) shot from grephene OS(an android derivative OS) and can't create thumbnail via expo-video/expo-video-thumbnail on IOS (26) either

other OS works fine, wonder if a iOS related parse issue

https://github.com/user-attachments/assets/1d0cd086-68f7-4a4d-aea4-f6f920c85468
